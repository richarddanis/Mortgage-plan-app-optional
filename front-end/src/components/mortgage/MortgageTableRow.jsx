import React from "react";
import { TableBody } from "@material-ui/core";
import MortgageTableRowView from "./MortgageTableRowView";

import PropTypes from 'prop-types';

const MortgageTableRow = ({list}) => {

  return (
      <>
        <TableBody>
          { list.map((row, index) => (<MortgageTableRowView key={index} sentence={row}/>)) }
        </TableBody>
      </>
  );
}

MortgageTableRow.propTypes = {
  list: PropTypes.array.isRequired,
};

export default MortgageTableRow;
