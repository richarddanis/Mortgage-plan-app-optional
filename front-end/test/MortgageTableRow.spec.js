import React from 'react';
import renderer from 'react-test-renderer';

import MortgageTableRow from "../src/components/mortgage/MortgageTableRow";

describe('MortgageTableRow', () => {
  it('renders correctly with items', () => {
    const itemList = ['test', 'test2']
    const tree = renderer.create(<MortgageTableRow list={itemList}/>).toJSON();

    expect(tree).toMatchSnapshot();
  })
})
