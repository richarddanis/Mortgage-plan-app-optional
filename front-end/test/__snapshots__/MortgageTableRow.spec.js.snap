// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MortgageTableRow renders correctly with items 1`] = `
<tbody
  className="MuiTableBody-root"
  role={null}
>
  <tr
    className="MuiTableRow-root"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
    >
      test
    </td>
  </tr>
  <tr
    className="MuiTableRow-root"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
    >
      test2
    </td>
  </tr>
</tbody>
`;
